src/main/java/se/sundsvall/seabloader/service/mapper/InvoiceMapper.java

```diff
@@ -57,4 +72,16 @@ public static InExchangeInvoiceStatusType toInExchangeInvoice(final String xml)
 		final var unmarshaller = JAXBContext.newInstance(InExchangeInvoiceStatusType.class).createUnmarshaller();
 		return (InExchangeInvoiceStatusType) JAXBIntrospector.getValue(unmarshaller.unmarshal(xmlSource));
 	}
+
+	private static InvoicePdf toInvoicePdf(final InExchangeInvoiceStatusType inExchangeInvoiceStatusType, final OutputStream outputStream) {
+
+		if (Objects.isNull(inExchangeInvoiceStatusType.getOriginalInvoice()) || Objects.isNull(outputStream)) {
+			LOGGER.warn("OriginalInvoice or attachments not found in invoice with invoiceId: {}", inExchangeInvoiceStatusType.getInvoice().getInvoiceId());
+			return null;
```

Comment:
As it is now,all requests will be sent, even those that have no attachments. Is it verified that InvoiceCache should be called even if attachments are missing?

An alternative is to throw an exception (causing the entity/dbpost to fail and be marked as EXPORT_FAILED in the DB) or that the service layer have logic to not send requests containing no attachments to InvoiceCache.

Reply:
I'll check this.

Reply:
This should result in an error according to Jakob. I'll fix.